### ISSUE TITLE
중간고사 이론 시험 6번 모범 답안

### ISSUE THREAD

**Issue by unizard** [sejongresearch/2020.Spring.AI#12]
- 문제
  - Linear regression과 Linear classification을 Hypothesis, Cost function으로 설명하시오.
- 정답 포인트
  - 각각의 가설, 비용함수 반드시 언급
  - 선형 회귀와 선형 분류를 올바르게 설명
  - 선형 분류의 경우 이진 혹은 다중으로 설명해도 무방하나 내용상의 틀린점은 있으면 안됨

**Comment by unizard** [comment]
// 18011817
Linear regression이란 수를 예측하는 회귀 문제라고 할 수 있다. 즉, 많은 데이터가 존재하고 그 데이터 사이의 관계를 찾아 숫자를 예측하는 문제이다. Linear regression을 문제를 풀기 위해 먼저 그 관계를 대표할 수 있는 Hypothesis(가설)을 설정해야한다. 이 때 Linear한 regression이기 때문에 가설을 일차 방정식인 H(x) = Wx+b 로 설정하자. W, b값에 따라 H(x)는 여러 값이 존재하는데, 이 중 선형회귀 모델을 대표할 수 있는 솔루션을 구하기 위해 예측값과 참값(Grade Truth)의 오차율이 가장 작은 것을 구한다. 이 오차율을 Cost function(Loss, Objective function)이라고 한다. cost function을 최소화하는 W, b를 찾음으로써 선형회귀 모델의 최적의 솔루션을 구하여, 최종적으로 데이터를 예측할 수 있게된다.
Linear classification이란 카테고리를 나누는 분류 문제라고 할 수 있다. Linear classification도 역시 Hypothesis(가설)을 세워 분류를 진행한다. 이 때 선형 회귀에서 사용하는 가설을 그대로 사용할 수 있는데(Logistic regression), 이 경우에는 데이터가 고르게 분포되어 있지 않을 때 (outline의 데이터로 인해) 분류의 정확도가 현저히 떨어지는 문제점이 발생한다. 따라서 선형 분류에서는 sigmoid, softmax 등의 함수를 이용하여 값을 0과 1 사이로 제한한다. 그 후 cost func. 역시 재설계하고 이를 최소화하는 W, b를 찾음으로써 선형 분류 모델의 최적의 솔루션을 구할 수 있게 된다

**Comment by unizard** [comment]
// 18011784
선형 회귀의 경우 다양한 데이터들 중에서 가장 그 데이터의 특성을 잘 나타내는하나의 직선을 찾는 것. 우리는 이 선을 찾기 위해 Hypothesis를 세우는데 선형회귀에서 Hypothesis는 y = WX + b 꼴로 나타난다. W와 b 값은 학습을 통해 갱신해 나가는데, 학습이 잘 되는지를 확인하는 방법은 cost 값이 줄어드는 것을 통해 확인할 수 있다. 선형회귀에서 cost fucntion은 (Hypothesis - y(정답값))^2을 평균으로 나타낸 것.

선형 분류는 말그대로 어떠한 class를 분류하는 작업을 말하는데, 기존 선형회귀 방법 (y = WX + b)으로는 분류를 할 수가 없다. 그래서 이를 해결하고자 sigmoid함수를 적용시켜 1/1-e^(WX+b) 꼴로 만들었는데 이것이 이진분류의 Hypothesis다. 그리고 이 때 cost function도 선형회귀의 cost function을 그대로 사용하면 미분값이 0이 되는 지점이 존재하기에, 이를 방지하고자 (-y*log(Hypothesis) - (1-y) * log(1-Hypothesis)).mean 형식으로 나타낸다.

다중 분류에 경우 이중분류에서 사용한 sigmoid를 사용하게 되면 확률값이 0~1 사이로 되지 않아 분류가 힘들다. 이를 해결하고자 다중분류에서는 softmax를 사용한다. softmax를 통해 output 값으로 나오는 class 확률의 총 합이 1로 나타날 수 있으며, 그 다음 argmax를 통해 제일 확률값이 높은 class를 1 나머지는 0으로 만드는 작업(onehot encoding)을 한다. 가설함수가 바뀌었으므로 cost function 역시 바뀌었는데 다중 분류에서 사용하는 cosf fucntion을 cross_entropy 라고 한다.

**Comment by unizard** [comment]
// 18011793

Linear regression은 주어진 데이터를 바탕으로 특정 정답 값을 예측하는 것이다. 그래서 hypothesis를 선형 방정식으로 선택(ex:H(x) = W*x + b)한다. Cost function은 오차 함수로써 예측된 값과 정답의 차이를 계산한다. //Linear classification은 범주형데이터를 encoding하여 사용하여 맞는 class를 찾는 분류를 하게 된다. binary classification을 예로 들자면 0과 1로 분류되어진다. 따라서 hypothesis를 0과 1사이로 bound 시킬 수 있는 sigmoid함수를 사용한다. Linear regression에서 사용하는 cost함수를 그대로 사용하면 지역극소점이 여러 곳 생겨 제대로된 학습이 불가능하다. 따라서 cost함수를 log함수를 통해서 사용한다. multi-variable classification에서는 cost함수를 cross-entropy함수를 통해 사용한다.

**Comment by unizard** [comment]
// 19011484

- Linear Regression
  - 입력을 이용하여 결과값을 예측하는 Task (예를들어, 온도와 습도, 수확량에 따른 배춧값 등)
  - Hypothesis는  𝑊𝑋+𝑏  꼴로 사용한다. Sigmoid 등을 씌우지 않는 이유는 값을 특정 범위로 제한하지 않기 위함이다.
  - Cost Function은 MSE나 RMSE 등을 사용한다.
- Linear Classification
  - 입력된 값이 어떤 클래스에 속하는지 분류하는 Task (입력된 사진 속 물체는 무엇인가?) 
  - 클래스가 두 개인 이진 분류와 두 개 이상인 다중 분류로 나뉜다.
  - 이진 분류의 경우,  𝑊𝑋+𝑏 에 Sigmoid 함수를 씌워, 출력값을 0~1로 제한하고 0에 가까우면 클래스 A, 1에 가까우면 클래스 B와 같은 형태로 해석한다.
  - 다중 분류의 경우,  𝑊𝑋+𝑏 에 softmax 함수를 씌워, 각 출력이 합이 1이 되는 0~1 사이의 값을 갖게 한 후, 각 클래스에 대한 출력을 입력이 해당 클래스일 확률로 해석한다.
  - Cost Function으로는 주로 Cross Entropy를 사용하는데, 이진 분류의 경우 Binary Cross Entropy를 써야 한다.
